reascripts/ReaSpeech/source/Widgets.lua

```diff
@@ -6,6 +6,24 @@
 
 Widgets = {}
 
+function Widgets.icon(icon, id, w, h, color)
+  color = color or 0xffffffff
+  local x, y = ImGui.GetCursorScreenPos(ctx)
+  local rv = ImGui.InvisibleButton(ctx, id, w, h)
+  local dl = ImGui.GetWindowDrawList(ctx)
+  icon(dl, x, y, w, h, color)
+  return rv
+end
+
+function Widgets.icon_button(icon, id, w, h, color)
```

Comment:
consider taking a tooltip string as an option to this and `icon`? they're paired enough throughout the code here to form a logical unit for my tastes. 🤔

a personal bugaboo is when i encounter UI iconography that isn't labeled and i have to interpret and predict what will happen like i'm an archaeologist looking at hieroglyphics trying to decide which rock to press so that the pyramid chamber door doesn't slam shut behind me forever

Reply:
I tried something like what you described, but I found that defining tooltip as its own function makes it more orthogonal. You can use it with any widget, whether it's one that's built into ImGui, one that we wrote ourselves, or one that hasn't been written yet. And you can change widgets without having to do anything different with the tooltip.

I concur that tooltips for icons are pretty essential from a usability perspective.

Reply:
> I tried something like what you described, but I found that defining tooltip as its own function makes it more orthogonal. You can use it with any widget, whether it's one that's built into ImGui, one that we wrote ourselves, or one that hasn't been written yet. And you can change widgets without having to do anything different with the tooltip.

i think we're mostly aligned here in thinking - `tooltip` remains its own function (useful in many contexts!), but that `icon`/`icon_button` would invoke it internally as well, instead of explicitly at each render site. extra benefit noticed when i talk it out like that is that it implicitly enforces the "no icons without discoverable labels." you'd have to go out of your way to supply an empty string like a jerk. 😂

> I concur that tooltips for icons are pretty essential from a usability perspective.

it literally makes me angry when i encounter an unexplained icon, especially when it's one i expect that i might have to click on